### ISSUE TITLE
How to plot waterfall plot for lightgbm

### ISSUE THREAD

**Issue by Wajih88** [shap/shap#2408]
In the documenation for Lightgbm, where shap values are computed in this way : 

```
explainer = shap.TreeExplainer(model)
shap_values = explainer.shap_values(X)
```

Where I try 
`shap.plots.waterfall(shap_values[0])`
I get this error 

`AttributeError: 'numpy.ndarray' object has no attribute 'base_values'`

**Comment by leliopusceddu** [comment]
i have the same problem

**Comment by HMY626** [comment]
Me too!

**Comment by Wajih88** [comment]
After some research this code worked for me (where `id` the row's number we want to explain)
```
explainer = shap.explainers.Tree(model)
shap_values = explainer(X)
shap.plots._waterfall.waterfall_legacy(explainer.expected_value[0],shap_values[id].values[:,0],
                                        feature_names=X.columns)
```
I hope this helps

**Comment by fonnesbeck** [comment]
This doesn't explain why this is happening. Why is `shap_values()` returning a numpy array when the plot functions don't expect a numpy array? Why do you have to use legacy functions?

**Comment by Wajih88** [comment]
After re examining my code, I believe this method is the correct one : 
```
# Object for plotting Water Fall
class helper_object():
    """
    This wraps the shap object.
    It takes as input i, which indicates the index of the observation to be explained.
    """
    def __init__(self, i):
        self.base_values = shap_values.base_values[i][predictions[i]]
        self.data = X.loc[i]
        self.feature_names = X.columns.to_list()
        self.values = shap_values.values[i][:,predictions[i]]
shap.plots.waterfall(helper_object(row_client),max_display=15)
```

**Comment by ori-n** [comment]
Here is how you can do it without a helper class:
https://stackoverflow.com/questions/66090750/how-to-show-feature-values-in-shap-waterfall-plot

**Comment by estellerambier** [comment]
> This doesn't explain why this is happening. Why is `shap_values()` returning a numpy array when the plot functions don't expect a numpy array? Why do you have to use legacy functions?


I am encountering the same exception with the `plots.heatmap`.
The plots are expecting a `shap.Explaination` object. Explicitly converting the output of the explainer solved it for me with 0.40.0:
```python
shap_values_explaination = shap.Explanation(shap_values, feature_names=trn.columns.tolist()) 
shap.plots.heatmap(shap_values_explaination)
```
